.codeclimate.yml

```diff
@@ -0,0 +1,162 @@
+# This is our default .CodeClimate.yml, broken out by language. Uncomment the
+# sections at the bottom that apply to your project. ACTION comments indicate
+# places where config might need to be tweaked.
+
+version: "2"
+
+plugins:
+
+# ---------------
+# Cross-language plugins. Should always be on.
+
+  duplication: # Looks for similar and identical code blocks
+    enabled: true
+    config:
+      languages:
+        go:
+        java:
+        javascript:
+        php:
+        python:
+          python_version: 3  # ACTION Comment this out if using Python 2
+        ruby:
+        swift:
+        typescript:
+
+  fixme: # Flags any FIXME, TODO, BUG, XXX, HACK comments so they can be fixed
+    enabled: true
+    config:
+      strings:
+      - FIXME
+      - TODO
+      - HACK
+      - XXX
+      - BUG
```

Comment:
Does this prevent merging PRs? If so, this is probably not what we want.

Reply:
Yes, it does (for newly added comments - old ones will be addressed as part of the cleanup). I'd argue that it should be what we want though. TODO comments work best for short term work. They're horrible for visualizing long term work. Making sure that master doesn't continue to accumulate invisible TODOs is why the check is on the merge action. Use them all you want while working on a branch, but when you're done anything left should be a ticket/backlog item so we can plan when and if to address it with the rest of our work.